### PR TITLE
ENH: added new properties on SeqView

### PR DESCRIPTION
**Pull request by GavinHuttley**
Plus two additional commits to deal with the impact of having no access to my GitHub repo for 14 days.

Ignore the merge and the commit related to `Alignment.get_degapped_relative_to()`.

### CODE REVIEW

File: tests/test_core/test_sequence.py
Comment (KatherineCaley) [inline tests/test_core/test_sequence.py:1]: explicitly test `absolute_start` and `absolute_stop` for cases where this is a `offset`?

At this point you are relying on the testing of of `annotation_offset` which now calls `absolute_start` to be complete.

File: src/cogent3/core/sequence.py [hunk 1]
```diff
@@ -1909,6 +1909,41 @@ def offset(self) -> int:
     def offset(self, value: int):
         self._offset = value or 0
 
+    @property
+    def absolute_start(self) -> int:
+        """returns the plus strand absolute start
```
Comment (KatherineCaley) [inline src/cogent3/core/sequence.py:1914]: Needs a more informative doctoring. It's complex for two reasons:
1. because "absolute" has alternate meanings in maths compared to general language. 
2.  because `absolute_start` is wrt to the underlying seq, unless there has been an offset set for the underlying annotations in which case it is wrt to the coordinates of the annotations.

File: src/cogent3/core/sequence.py [hunk 2]
```diff
@@ -1909,6 +1909,41 @@ def offset(self) -> int:
...
+            assert self.stop < 0, "expected stop on reverse strand SeqView < 0"
+            start = self.stop + len(self.seq) + 1
+        else:
+            start = self.start
+
+        return self.offset + start
+
+    @property
+    def absolute_stop(self) -> int:
+        """returns the plus strand absolute stop
```
Comment (KatherineCaley) [inline src/cogent3/core/sequence.py:1932]: see previous comment on adding a bit of detail to docstring

File: tests/test_core/test_sequence.py [hunk 3]
```diff
@@ -2381,6 +2381,27 @@ def test_sliced_seqview_rich_dict(reverse):
     assert rd["offset"] == 2
 
 
+@pytest.mark.parametrize(
+    "sl",
+    (
+        slice(2, 5, 1),  # positive indices, positive step
```
Comment (KatherineCaley) [inline tests/test_core/test_sequence.py:2387]: would be good to test edge cases, especially for slices at the ends of the sequence.
- if you slice the whole sequence (or the whole seq in reverse), do you get 0 for start, and the length of sequence for end?
- singleton slices, and singleton slices at the start and ends (and reversed).

**Review by KatherineCaley — CHANGES_REQUESTED** [review]

**Comment by KatherineCaley** [comment]
Looks good! some small changes requested for the documentation and testing.

File: src/cogent3/core/sequence.py (hunk 1 above)
Comment (GavinHuttley) [reply src/cogent3/core/sequence.py:1914]: Good points. Changing the name is probably best bet. Which of the following seems best
- `parent_start`
- `full_start`
- `resolved_start`
- `adjusted_start`
or something else?

File: tests/test_core/test_sequence.py (hunk 3 above)
Comment (GavinHuttley) [reply tests/test_core/test_sequence.py:2387]: by singleton slice do you mean `slice(i, i+1)`?

Comment (KatherineCaley) [reply tests/test_core/test_sequence.py:2387]: yep

File: src/cogent3/core/sequence.py (hunk 1 above)
Comment (KatherineCaley) [reply src/cogent3/core/sequence.py:1914]: I think `parent_start`

**Review by KatherineCaley — APPROVED** [review]
